src/lib.rs

```diff
@@ -419,12 +424,54 @@ impl EvmContract {
 
         assert!(output.len() == 32, "Invalid ECDSA signature.");
 
+        let signer = Address::from_slice(&output[12..32]);
+        if self.revoked_aliases.contains(&signer[..].to_vec()) || self.ecrecover_aliases.get(&signer[..].to_vec()).is_some() {
+            return false;
+        }
+
         self.set_ecrecover_alias(
-            &Address::from_slice(&output[12..32]),
+            &signer,
             &utils::predecessor_as_evm(),
         );
         true
     }
+
+
+    /// Allow a Near account to permanently revoke an ECDSA alias it controls. The Near account
+    /// must have previously registered the alias, and it must still be active (not previously
+    /// revoked).
+    ///
+    /// Once an ECDSA alias is revoked, it may NEVER be used to alias any account again. This is a
+    /// permanent, global, revocation.
+    ///
+    /// # Arguments
+    ///
+    /// * `signature` - a valid ethereum-formatted signature
+    ///
+    /// # Returns
+    ///
+    /// True if succesfully revoked False if the key was never registered.
+    ///
+    /// # Warning
+    ///
+    /// This CANNOT be undone. The key can NEVER be aliased again.
+    ///
+    /// # Panics
+    ///
+    /// - When alias_address is not a valid EVM address
+    pub fn revoke_ecdsa_alias(&mut self, alias_address: String) -> bool {
+        let sender = utils::predecessor_as_evm();
+        let alias = utils::hex_to_evm_address(&alias_address);
+
+        if self.get_ecrecover_alias(alias) == Some(sender) {
+            self.set_ecrecover_alias(&alias, &alias);
```

Comment:
if evm alias == evm alias means it's revoked? no zero val?

Reply:
I did this so that an alias can never be re-registered by another near account. Setting to 0 value would allow re-use of the alias in potentially unsafe ways

Reply:
ahhh ok